src/vcpkg/base/files.cpp

```diff
@@ -2208,35 +2259,197 @@ namespace vcpkg
+        {
+            if (mkdir(new_directory, 0777) == 0)
+            {
+                return -1;
+            }
+
+            auto last_error = errno;
+            if (last_error == EEXIST)
+            {
+                struct stat s;
+                if (stat(new_directory, &s) == 0)
+                {
+                    if (S_ISDIR(s.st_mode))
+                    {
+                        return 0;
+                    }
+                }
+                else
+                {
+                    return errno;
+                }
+            }
+
+            return last_error;
+        }
+#endif // ^^^ !_WIN32
+
         virtual bool create_directory(const Path& new_directory, std::error_code& ec) override
         {
+#if defined(_WIN32)
             return stdfs::create_directory(to_stdfs_path(new_directory), ec);
+#else  // ^^^ _WIN32 // !_WIN32 vvv
+            auto attempt = posix_create_directory(new_directory.c_str());
+            if (attempt <= 0)
+            {
+                ec.clear();
+                return attempt != 0;
+            }
+
+            ec.assign(attempt, std::generic_category());
+            return false;
+#endif // ^^^ !_WIN32
         }
         virtual bool create_directories(const Path& new_directory, std::error_code& ec) override
         {
+#if defined(_WIN32)
             return stdfs::create_directories(to_stdfs_path(new_directory), ec);
+#else // ^^^ _WIN32 // !_WIN32 vvv
+            ec.clear();
+            if (new_directory.empty())
+            {
+                return false;
+            }
+
+            const auto& new_str = new_directory.native();
+            auto first = new_str.begin();
+            const auto last = new_str.end();
+            std::string this_create;
+            // establish the !is_slash(*first) loop invariant
+            if (is_slash(*first))
+            {
+                this_create.push_back('/');
+                first = std::find_if_not(first, last, is_slash);
+            }
+
+            bool last_mkdir_created = false;
+            for (;;)
+            {
+                if (first == last)
+                {
+                    return last_mkdir_created;
+                }
+
+                assert(!is_slash(*first));
+                const auto next_slash = std::find_if(first, last, is_slash);
+                this_create.append(first, next_slash);
+                const auto attempt = posix_create_directory(this_create.c_str());
+                if (attempt > 0)
+                {
+                    ec.assign(attempt, std::generic_category());
+                    return false;
+                }
+
+                last_mkdir_created = attempt != 0;
+                if (next_slash == last)
+                {
+                    return last_mkdir_created;
+                }
+
+                this_create.push_back('/');
+                first = std::find_if_not(next_slash, last, is_slash);
```

Comment:
Could you add a comment here about this existing for purposes of multiple slashes in a row?